### PR TITLE
rdtsc_sync_test:add new case to check rdtsc sync

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 2087599

The behaviour of rdtsc should be same
after rebooting.

Signed-off-by: nanliu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
(1/1) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.rdtsc_sync_test.q35: PASS (12.57 s)

# cat /tmp/test_file 
enabling apic
PM timer port is 608
rdtsc = 3110683200
rdtsc = 6104995710
rdtsc = 9099299580
rdtsc = 12093613770
rdtsc = 15087918810
enabling apic
PM timer port is 608
rdtsc = 3078688201
rdtsc = 6072997081
rdtsc = 9067308151
rdtsc = 12061616881

**Comment by nanliu-r** [comment]
@PaulYuuu Could you please help to review this patch? Thanks a lot!

File: qemu/tests/cfg/rdtsc_sync_test.cfg [hunk 1]
```diff
@@ -0,0 +1,7 @@
+- rdtsc_sync_test:
+    type = rdtsc_sync_test
+    start_vm = no
+    auto_cpu_policy = "virttest"
+    source_file = 'rdtsc_sync_test/bz1975840.flat'
```
Comment (PaulYuuu) [inline qemu/tests/cfg/rdtsc_sync_test.cfg:5]: Using bug ID is not a good idea, the file should have a clear name to explain what it does.

File: qemu/tests/rdtsc_sync_test.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    process.run('echo -e \'{"execute":"qmp_capabilities"}'
+                '{"execute":"system_reset"}\'|nc -U /tmp/mm',
+                shell=True, verbose=True)
+    time.sleep(5)
+    process.run('echo -e \'{"execute":"qmp_capabilities"}'
+                '{"execute":"quit"}\'|nc -U /tmp/mm',
+                shell=True, verbose=True)
+
+    qemu_pid = process.getoutput("pgrep -f %s" % src_test_binary, shell=True)
+    if not qemu_pid:
```
Comment (PaulYuuu) [inline qemu/tests/rdtsc_sync_test.py:49]: Seems it's a typo. The qemu process is quit, so here is qemu_pid still existed, it should fail.

Comment (PaulYuuu) [reply qemu/tests/rdtsc_sync_test.py:49]: Also, you can use the exit status, when no qemu process, `pgrep -f` returns `1`.

File: qemu/tests/rdtsc_sync_test.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        test.fail("QEMU quit failed!")
+    is_file = os.path.exists(log_file)
+    if not is_file:
+        test.fail("Can't find the log file %s" % log_file)
+
+    value_a = []
+    value_b = []
+    reset = False
+    try:
+        f = open(log_file, "r")
```
Comment (PaulYuuu) [inline qemu/tests/rdtsc_sync_test.py:59]: It is safer to use `with open("log_file")`.

File: qemu/tests/cfg/rdtsc_sync_test.cfg [hunk 4]
```diff
@@ -0,0 +1,7 @@
+- rdtsc_sync_test:
+    type = rdtsc_sync_test
+    start_vm = no
```
[the remark this answers is not in the document]
Comment (nanliu-r) [reply qemu/tests/cfg/rdtsc_sync_test.cfg]: thanks and updated

File: qemu/tests/cfg/rdtsc_sync_test.cfg (hunk 1 above)
Comment (nanliu-r) [reply qemu/tests/cfg/rdtsc_sync_test.cfg]: > Using bug ID is not a good idea, the file should have a clear name to explain what it does.

Thanks and updated.

File: qemu/tests/rdtsc_sync_test.py (hunk 2 above)
Comment (nanliu-r) [reply qemu/tests/rdtsc_sync_test.py]: > Also, you can use the exit status, when no qemu process, `pgrep -f` returns `1`.

Thanks and updated, and I will keep current style to be same as above.

**Comment by nanliu-r** [comment]
@PaulYuuu please help to review this again, thanks.

(1/1) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.rdtsc_sync_test.q35: PASS (12.23 s)

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**